### PR TITLE
Fix WL-2961

### PR DESCRIPTION
**Pull request by ghost**
Some refactoring and introduction of the callback method for oxitems.

### CODE REVIEW

File: library/src/webapp/js/headscripts.js [hunk 1]
```diff
@@ -704,3 +708,8 @@ function rewriteYouTubeEmbeds() {
     }
 
 }
+
+// the function to call when oxitems has finished loading
+function oxitemsCallback() {
+    fixLinksForMixedContent();
```
Comment (buckett) [inline library/src/webapp/js/headscripts.js:717]: Do we want todo this when we are not in an iframe?

File: library/src/webapp/js/headscripts.js [hunk 2]
```diff
@@ -645,32 +645,27 @@ function addEvent(element, event, fn) {
...
+function fixMixedContentReferences() {
+    rewriteYouTubeEmbeds();
+    if (window.top != window.self) {
+        // I am in an iframe
+        fixLinksForMixedContent();
+    }
+}
 
-                rewriteWebLearnHref(link);
+function fixLinksForMixedContent() {
```
Comment (buckett) [inline library/src/webapp/js/headscripts.js:661]: Documentation on method.

File: library/src/webapp/js/headscripts.js (hunk 1 above)
Comment (ghost) [inline library/src/webapp/js/headscripts.js:717]: No I don't think so. However there is a good argument to adjust old WebLearn links regardless.

**Comment by ghost** [comment]
I've made it only rewrite links when in a frame.

**Comment by ghost** [comment]
Fixed

**Pull request merged**